datareduction/datareducer.cpp

```diff
@@ -538,8 +538,36 @@ void initializeDataReducers(DataReducer * outputReducer, DataReducer * diagnosti
          ));
          continue;
       }
-      if(*it == "fg_HallE") {
-         // TODO: Add outputreducer to get EHALL from fsgrid
+      if(*it == "HallE" || *it == "fg_HallE") {
+         for(int index=0; index<fsgrids::N_EHALL; index++) {
+            std::string reducer_name = "fg_HallE" + std::to_string(index);
+            outputReducer->addOperator(new DRO::DataReductionOperatorFsGrid(reducer_name,[index](
+                         FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2>& perBGrid,
+                         FsGrid< std::array<Real, fsgrids::efield::N_EFIELD>, 2>& EGrid,
+                         FsGrid< std::array<Real, fsgrids::ehall::N_EHALL>, 2>& EHallGrid,
+                         FsGrid< std::array<Real, fsgrids::egradpe::N_EGRADPE>, 2>& EGradPeGrid,
+                         FsGrid< std::array<Real, fsgrids::moments::N_MOMENTS>, 2>& momentsGrid,
+                         FsGrid< std::array<Real, fsgrids::dperb::N_DPERB>, 2>& dPerBGrid,
+                         FsGrid< std::array<Real, fsgrids::dmoments::N_DMOMENTS>, 2>& dMomentsGrid,
+                         FsGrid< std::array<Real, fsgrids::bgbfield::N_BGB>, 2>& BgBGrid,
+                         FsGrid< std::array<Real, fsgrids::volfields::N_VOL>, 2>& volGrid,
+                         FsGrid< fsgrids::technical, 2>& technicalGrid)->std::vector<double> {
+
+                  std::array<int32_t,3>& gridSize = technicalGrid.getLocalSize();
+                  std::vector<double> retval(gridSize[0]*gridSize[1]*gridSize[2]);
+
+                  // Iterate through fsgrid cells and extract EHall
+                  for(int z=0; z<gridSize[2]; z++) {
```

Comment:
Inconsistent indentation? Up to you, Mr Lead Developer. :)

Reply:
Our coding style guidelines are anyway kinda fishy about how to indent lambdas properly. But I did change the for indentation to three spaces now. Thanks!